sopel/tools/__init__.py

```diff
@@ -223,6 +223,7 @@ def compile_rule(nick, pattern, alias_nicks):
     return re.compile(pattern, flags)
 
 
+@deprecated('regexp functions are now private', '7.1', '8.0')
```

Comment:
I'm gonna wordsmith a bit in this file. Thought about just making the suggestion once, but it's probably easier on you if I suggest it everywhere so you can just batch-apply the change if you like it. 😁

```suggestion
@deprecated('command regexp tools are now private', '7.1', '8.0')
```